### PR TITLE
port counter support on sonic-vs

### PR DESCRIPTION
**Pull request by vishnushetty**
This is continuation of previous PR.

https://github.com/sonic-net/sonic-sairedis/pull/1274

In this PR addressed review comments. Please review.

### CODE REVIEW

File: vslib/SwitchState.cpp [hunk 1]
```diff
@@ -17,6 +17,25 @@ using namespace saivs;
...
+        { SAI_PORT_STAT_IF_IN_DISCARDS, "rx_dropped" },
+        { SAI_PORT_STAT_IF_OUT_OCTETS, "tx_bytes" },
+        { SAI_PORT_STAT_IF_OUT_UCAST_PKTS, "tx_packets" },
+        { SAI_PORT_STAT_IF_OUT_ERRORS, "tx_errors" },
+        { SAI_PORT_STAT_IF_OUT_DISCARDS, "tx_dropped" }
+    };
+
+    return statIdMap;
+}
+const std::map<sai_stat_id_t, std::string> SwitchState::m_statIdMap = SwitchState::createStatIdMap();
```
Comment (kcudnik) [inline vslib/SwitchState.cpp:37]: i thought you will assign this map directly but if you want to do it as function, so let it be

File: vslib/SwitchState.cpp [hunk 2]
```diff
@@ -199,6 +218,79 @@ void SwitchState::asyncOnLinkMsg(
...
+    {
+        counter = -1;
+        SWSS_LOG_ERROR("get counter cmd %s failed fp=%x", cmd, fp);
+        rc = SAI_STATUS_FAILURE;
+    }
+
+    if (fp != NULL)
+    {
+        pclose(fp);
+    }
```
Comment (kcudnik) [inline vslib/SwitchState.cpp:261]: as proposed, works for me:

```c++
#include <fstream>
#include <iostream>
#include <string>

// $ g++ test.cpp -o test
// $ ./test
// 40515

int main()
{

    std::string s = "/sys/class/net/lo/statistics/rx_bytes";

    std::ifstream ifs (s.c_str(), std::ifstream::in);

    uint64_t stat;

    if (ifs.good())
    {
        ifs >> stat;

        std::cout << stat << std::endl;
    }

    return 0;
}
```

Comment (vishnushetty) [reply vslib/SwitchState.cpp]: ok. May be "std::ios::in" or read was an issue.

**Comment by kcudnik** [comment]
please fix build errors

**Comment by vishnushetty** [comment]
sonic-sairedis is successful. Build error is in orchagent. How to refresh other repo and build?

In file included from dash/dashaclorch.h:19,
                 from orchdaemon.h:49,
                 from main.cpp:25:
dash/dashorch.h:21:10: fatal error: dash_api/appliance.pb.h: No such file or directory
   21 | #include "dash_api/appliance.pb.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[4]: *** [Makefile:905: orchagent-main.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: Leaving directory '/__w/1/s/orchagent'

**Comment by kcudnik** [comment]
@Pterosaur hey, you have been working on dash_api/ in swss repo, can you fix pipeline here on sairedis ? https://github.com/sonic-net/sonic-swss/pull/2722 seems like your PR added some some libs libdashapi

**Comment by kcudnik** [comment]
maybe adding this: https://github.com/sonic-net/sonic-swss/pull/2722/files#diff-437c0ec0e84bd6e28f53683d7df7355c6d304d1049126607e49416df45d73c0aR162 to this https://github.com/sonic-net/sonic-sairedis/blob/master/.azure-pipelines/build-swss-template.yml#L111 will be good enough

**Comment by kcudnik** [comment]
my attempt to fix: https://github.com/sonic-net/sonic-sairedis/pull/1276, this will probably fail for the same reason, since az pipeline needs to be checked in first to take effect

**Comment by kcudnik** [comment]
ok, seems like my attempt fixed swss build,  test failure not related to build

**Comment by kcudnik** [comment]
build was fixed by https://github.com/sonic-net/sonic-sairedis/pull/1277, let restart

/azp/run

**Comment by kcudnik** [comment]
nice error XD
```
[       OK ] ClientServerSai.FDB_ENTRY (10 ms)
[ RUN      ] ClientServerSai.INSEG_ENTRY
Assertion failed: pfd.revents & POLLIN (src/signaler.cpp:265)
/bin/bash: line 6: 23663 Aborted                 (core dumped) ${dir}$tst

```

File: vslib/SwitchState.cpp [hunk 3]
```diff
@@ -199,6 +211,65 @@ void SwitchState::asyncOnLinkMsg(
...
 }
 
+sai_status_t SwitchState::getNetStat(
+        _In_ sai_stat_id_t counterId,
+        _In_ std::string& ifName,
+        _Out_ uint64_t& counter)
+{
+    SWSS_LOG_ENTER();
+
+    std::string counterName;
```
Comment (kcudnik) [inline vslib/SwitchState.cpp:221]: not used variable

File: vslib/SwitchState.cpp [hunk 4]
```diff
@@ -199,6 +211,65 @@ void SwitchState::asyncOnLinkMsg(
...
+    std::string ifName;
+
+    /* zero out counter */
+    counter = 0;
+
+    if (getTapNameFromPortId(portId, ifName) == false)
+    {
+        /* Port not ready */
+        SWSS_LOG_DEBUG("Port not ready %s", sai_serialize_object_id(portId).c_str());
+        return SAI_STATUS_SUCCESS;
```
Comment (kcudnik) [inline vslib/SwitchState.cpp:261]: why this returns success on port get failure ?

File: vslib/SwitchState.cpp (hunk 3 above)
Comment (vishnushetty) [reply vslib/SwitchState.cpp:221]: ok

**Review by kcudnik — APPROVED** [review]

**Comment by kcudnik** [comment]
please change pr name to more meaningful

**Comment by skg-net** [comment]
@vishnushetty Can you please update the Quality Metric (Alpha/Beta/GA) for the feature either in this PR comments or in HLD itself based on https://github.com/sonic-net/SONiC/blob/master/doc/SONiC%20feature%20quality%20definition.md
Thanks

**Pull request merged**